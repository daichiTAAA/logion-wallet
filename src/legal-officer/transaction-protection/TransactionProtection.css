.TransactionProtection .void-locs {
    margin-top: 30px;
}
