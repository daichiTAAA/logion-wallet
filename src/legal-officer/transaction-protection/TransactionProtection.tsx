import React, { useState } from 'react';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';

import { FullWidthPane } from '../../common/Dashboard';
import Tabs from '../../common/Tabs';

import { useCommonContext } from '../../common/CommonContext';
import PendingLocRequests from './PendingLocRequests';
import RejectedLocRequests from './RejectedLocRequests';
import OpenedLocs from './OpenedLocs';
import ClosedLocs from './ClosedLocs';
import Frame from '../../common/Frame';

import VoidLocs from './VoidLocs';

export default function TransactionProtection() {
    const { colorTheme } = useCommonContext();
    const [ locTabKey, setLocTabKey ] = useState<string>('open');
    const [ requestTabKey, setRequestTabKey ] = useState<string>('pending');

    return (
        <FullWidthPane
            mainTitle="Transaction Protection Cases"
            titleIcon={ {
                icon: {
                    id: 'loc'
                },
                background: colorTheme.topMenuItems.iconGradient,
            } }
            className="TransactionProtection"
        >
            <Row>
                <Col>
                    <Frame
                        title="Transaction Protection Case(s)"
                    >
                        <Tabs
                            activeKey={ locTabKey }
                            onSelect={ key => setLocTabKey(key || 'open') }
                            tabs={[
                                {
                                    key: "open",
                                    title: "Open",
                                    render: () => <OpenedLocs locType="Transaction" />
                                },
                                {
                                    key: "closed",
                                    title: "Closed",
                                    render: () => <ClosedLocs locType="Transaction" />
                                },
                                {
                                    key: "void",
                                    title: "Void",
                                    render: () => <VoidLocs locType="Transaction" />
                                }
                            ]}
                        />
                    </Frame>
                </Col>
            </Row>
            <Row>
                <Col>
                    <Frame
                        title="Transaction Protection Case Request(s)"
                    >
                        <Tabs
                            activeKey={ requestTabKey }
                            onSelect={ key => setRequestTabKey(key || 'pending') }
                            tabs={ [
                                {
                                    key: "pending",
                                    title: "Pending",
                                    render: () => <PendingLocRequests />
                                },
                                {
                                    key: "rejected",
                                    title: "Rejected",
                                    render: () => <RejectedLocRequests />
                                }
                            ]}
                        />
                    </Frame>
                </Col>
            </Row>
        </FullWidthPane>
    );
}
