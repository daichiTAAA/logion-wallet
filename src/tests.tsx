import { ReactElement } from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import reactRenderer, { ReactTestRenderer, ReactTestRendererJSON, act as reactAct } from 'react-test-renderer';

export function render(element: ReactElement): null | ReactTestRendererJSON | ReactTestRendererJSON[] {
    const component = reactRenderer.create(element);
    return component.toJSON();
}

export function renderer(element: ReactElement): ReactTestRenderer {
    return reactRenderer.create(element);
}

export function shallowRender<E extends ReactElement>(element: ReactElement): E {
    const renderer = createRenderer();
    renderer.render(element);
    return renderer.getRenderOutput();
}

export function mockAccount(address: string, name: string) {
    return {
        address,
        meta: {
            name
        }
    };
}

export { reactAct as act };
