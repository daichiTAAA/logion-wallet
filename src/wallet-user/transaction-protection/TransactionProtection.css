.TransactionProtection .action-bar {
    margin-top: 30px;
}

.TransactionProtection .closed-void-locs {
    margin-top: 30px;
}
