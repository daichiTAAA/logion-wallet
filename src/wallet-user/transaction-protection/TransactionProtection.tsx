import React, { useState } from 'react';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';

import { FullWidthPane } from '../../common/Dashboard';
import Tabs from '../../common/Tabs';
import Frame from '../../common/Frame';
import { useCommonContext } from "../../common/CommonContext";

import OpenedLocs from './OpenedLocs';
import RequestedLocs from './RequestedLocs';
import RejectedLocs from './RejectedLocs';
import LocCreation from './LocCreation';
import ClosedLocs from './ClosedLocs';

import './TransactionProtection.css';

export default function TransactionProtection() {
    const { colorTheme } = useCommonContext();
    const [ locTabKey, setLocTabKey ] = useState<string>('open');
    const [ requestTabKey, setRequestTabKey ] = useState<string>('pending');

    return (
        <FullWidthPane
            mainTitle="Transaction Protection"
            subTitle="Logion Officer Cases"
            titleIcon={{
                icon: {
                    id: 'loc'
                },
                background: colorTheme.topMenuItems.iconGradient,
            }}
            className="TransactionProtection"
        >
            <Row>
                <Col>
                    <Frame
                        title="Transaction Protection Case(s)"
                    >
                        <Tabs
                            activeKey={ locTabKey }
                            onSelect={ key => setLocTabKey(key || 'open') }
                            tabs={[
                                {
                                    key: "open",
                                    title: "Open",
                                    render: () => <OpenedLocs />
                                },
                                {
                                    key: "closed",
                                    title: "Closed",
                                    render: () => <ClosedLocs />
                                }
                            ] }
                        />
                        <div className="action-bar">
                            <LocCreation />
                        </div>
                    </Frame>
                </Col>
            </Row>
            <Row>
                <Col>
                    <Frame
                        title="Transaction Protection Request(s)"
                    >
                        <Tabs
                            activeKey={ requestTabKey }
                            onSelect={ key => setRequestTabKey(key || 'pending') }
                            tabs={[
                                {
                                    key: "pending",
                                    title: "Pending",
                                    render: () => <RequestedLocs/>
                                },
                                {
                                    key: "rejected",
                                    title: "Rejected",
                                    render: () => <RejectedLocs/>
                                }
                            ]}
                        />
                    </Frame>
                </Col>
            </Row>
        </FullWidthPane>
    );
}
