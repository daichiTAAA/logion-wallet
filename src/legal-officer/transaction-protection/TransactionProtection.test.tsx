jest.mock('../../common/CommonContext');

import React from 'react';
import TransactionProtection from './TransactionProtection';
import { shallowRender } from '../../tests';

test('renders', () => {
    const tree = shallowRender(<TransactionProtection/>);
    expect(tree).toMatchSnapshot();
});
