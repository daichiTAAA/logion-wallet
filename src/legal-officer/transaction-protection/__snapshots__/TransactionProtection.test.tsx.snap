// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<FullWidthPane
  className="TransactionProtection"
  mainTitle="Transaction Protection Cases"
  titleIcon={
    Object {
      "background": Object {
        "from": "#3b6cf4",
        "to": "#6050dc",
      },
      "icon": Object {
        "id": "loc",
      },
    }
  }
>
  <Row>
    <Col>
      <Frame
        title="Transaction Protection Case(s)"
      >
        <Tabs
          activeKey="open"
          onSelect={[Function]}
          tabs={
            Array [
              Object {
                "key": "open",
                "render": [Function],
                "title": "Open",
              },
            ]
          }
        />
        <Tabs
          activeKey="closed"
          className="void-locs"
          onSelect={[Function]}
          tabs={
            Array [
              Object {
                "key": "closed",
                "render": [Function],
                "title": "Closed",
              },
              Object {
                "key": "void",
                "render": [Function],
                "title": "Void",
              },
            ]
          }
        />
      </Frame>
    </Col>
  </Row>
  <Row>
    <Col>
      <Frame
        title="Transaction Protection Case Request(s)"
      >
        <Tabs
          activeKey="pending"
          onSelect={[Function]}
          tabs={
            Array [
              Object {
                "key": "pending",
                "render": [Function],
                "title": "Pending",
              },
              Object {
                "key": "rejected",
                "render": [Function],
                "title": "Rejected",
              },
            ]
          }
        />
      </Frame>
    </Col>
  </Row>
</FullWidthPane>
`;
